### DISCUSSION TITLE
Refresh Token Scenario

### DISCUSSION THREAD

**Discussion by sadegh-khan** [saleor/saleor#13496] [Q&A]
i want to know what is the refreshToken scenario in saleor.
Why with every page refresh, 
1) page freeze for 5 seconds (from front)
2) call "refreshTokenWithUser"
why for each refresh (for example under 10 second page refresh) we need to call "refreshTokenWithUser" and for doing that we freeze page for about 5 seconds??

**Comment by zaiste** [comment]
@sadegh-khan could you provide a bit more details about your use case? Is your frontend custom built? Do you use our legacy SDK ([saleor/saleor-sdk](https://github.com/saleor/saleor-sdk)) or the new one ([saleor/auth-sdk](https://github.com/saleor/auth-sdk)) when communicating with the API?

**Reply by sadegh-khan** [reply]
my front is custom built ,
for more detail to build my front , i follow  "saleor-dashboard" and " react-storefront", so there i see that with every page reload, front call "refreshTokenWithUser", and for avoiding other api call, freeze for 5 seconds, alson dont save any cookie!
